### PR TITLE
DIV-5908: Moving more dependencies into div-common-lib

### PR DESCRIPTION
**Pull request by jvictorcf**
### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: config/checkstyle/checkstyle.xml [hunk 1]
```diff
@@ -46,7 +45,7 @@
       <property name="allowNonPrintableEscapes" value="true"/>
     </module>
     <module name="LineLength">
-      <property name="max" value="120"/>
+      <property name="max" value="150"/>
```
Comment (jacmurphy) [inline config/checkstyle/checkstyle.xml:48]: Will this case any issues with styling ?

File: src/main/java/uk/gov/hmcts/reform/divorce/mapper/MappingCommons.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        }
+        return BooleanUtils.toStringYesNo(BooleanUtils.toBoolean(value)).toUpperCase(Locale.ENGLISH);
+    }
+
+    public static String toYesNoNeverPascalCase(final String value) {
+        if (Strings.isNullOrEmpty(value)) {
+            return null;
+        }
+        return YesNoNeverAnswer.fromInput(value).getAnswer();
+    }
```
Comment (jacmurphy) [inline src/main/java/uk/gov/hmcts/reform/divorce/mapper/MappingCommons.java:48]: When would something like this be used?

File: src/main/java/uk/gov/hmcts/reform/divorce/mapper/strategy/reasonfordivorce/ReasonForDivorceStrategy.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+    String deriveStatementOfCase(DivorceSession divorceSession);
+
+    default void setLivedApartFieldsFromDivorceSession(DivorceSession divorceSession, CoreCaseData coreCaseData) {
+        //only applicable for some divorce reasons
+    }
+
+    default void setLivedApartFieldsFromCoreCaseData(CoreCaseData coreCaseData, DivorceSession divorceSession) {
+        //only applicable for some divorce reasons
+    }
```
Comment (jacmurphy) [inline src/main/java/uk/gov/hmcts/reform/divorce/mapper/strategy/reasonfordivorce/ReasonForDivorceStrategy.java:18]: Did you mean to include these ?

File: src/main/java/uk/gov/hmcts/reform/divorce/mapper/strategy/reasonfordivorce/UnreasonableBehaviourStrategy.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.divorce.model.usersession.DivorceSession;
+
+@Component
+public class UnreasonableBehaviourStrategy implements ReasonForDivorceStrategy {
+
+    private static final String UNREASONABLE_BEHAVIOUR = "unreasonable-behaviour";
+    private static final String LINE_SEPARATOR = "\n";
```
Comment (jacmurphy) [inline src/main/java/uk/gov/hmcts/reform/divorce/mapper/strategy/reasonfordivorce/UnreasonableBehaviourStrategy.java:11]: usage again

File: src/main/java/uk/gov/hmcts/reform/divorce/service/impl/SeparationDateServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+public class SeparationDateServiceImpl implements SeparationDateService {
+
+    @Override
+    public void updateSeparationDate(DivorceSession divorceSession) {
+        Date separationDate = divorceSession.getReasonForDivorceDecisionDate();
+
+        if (separationDate == null
+            || divorceSession.getReasonForDivorceLivingApartDate() != null
+            && separationDate.before(divorceSession.getReasonForDivorceLivingApartDate())) {
+            separationDate = divorceSession.getReasonForDivorceLivingApartDate();
```
Comment (jacmurphy) [inline src/main/java/uk/gov/hmcts/reform/divorce/service/impl/SeparationDateServiceImpl.java:19]: Could this be split out into a readable private function?

File: src/main/java/uk/gov/hmcts/reform/divorce/utils/Constants.java [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class Constants {
+
+    public static final String YES_VALUE = "Yes";
+    public static final String NO_VALUE = "No";
+
```
Comment (jacmurphy) [inline src/main/java/uk/gov/hmcts/reform/divorce/utils/Constants.java:11]: Maybe add the "\n" here?

File: src/main/java/uk/gov/hmcts/reform/divorce/mapper/MappingCommons.java (hunk 2 above)
Comment (jvictorcf) [reply src/main/java/uk/gov/hmcts/reform/divorce/mapper/MappingCommons.java:48]: Just transferring the classes today.

File: src/main/java/uk/gov/hmcts/reform/divorce/mapper/strategy/reasonfordivorce/ReasonForDivorceStrategy.java (hunk 3 above)
Comment (jvictorcf) [reply src/main/java/uk/gov/hmcts/reform/divorce/mapper/strategy/reasonfordivorce/ReasonForDivorceStrategy.java:18]: Just moved the classes

File: src/main/java/uk/gov/hmcts/reform/divorce/mapper/MappingCommons.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+import com.google.common.base.Strings;
+import org.apache.commons.lang3.BooleanUtils;
+import uk.gov.hmcts.reform.divorce.model.usersession.YesNoNeverAnswer;
+
+import java.util.Locale;
+import java.util.Objects;
+
+public final class MappingCommons {
+
+    public static final String SIMPLE_DATE_FORMAT = "yyyy-MM-dd";
```
Comment (yarpo) [inline src/main/java/uk/gov/hmcts/reform/divorce/mapper/MappingCommons.java:13]: Are you sure we need it, I'd use:

```
uk.gov.hmcts.reform.divorce.utils.DateUtils.Formats#CCD_DATE
```

**Pull request merged**